packages/cli/lib/daemon/seeder.js

```diff
@@ -0,0 +1,147 @@
+import Hyperswarm from 'hyperswarm'
+import Corestore from 'corestore'
+import * as SlashURL from '@synonymdev/slashtags-url'
+import HyperDrive from 'hyperdrive'
+import goodbye from 'graceful-goodbye'
+
+import { SEEDER_STORE_DIRECTORY } from '../constants.js'
+
+const SEEDERS_TOPIC = Buffer.from('3b9f8ccd062ca9fc0b7dd407b4cd287ca6e2d8b32f046d7958fa7bea4d78fd75', 'hex')
+
+export default class Seeder {
+  /**
+   * @param {import('@hyperswarm/dht')} dht
+   * @param {import('level').Level<string, object>} db
+   */
+  constructor (dht, db) {
+    this.swarm = new Hyperswarm({ dht })
+    this.corestore = new Corestore(SEEDER_STORE_DIRECTORY)
+
+    this.swarm.on('connection', (conn, peerInfo) => {
+      console.log('Got peer connection', peerInfo.publicKey.toString('hex'))
+      this.corestore.replicate(conn)
+    })
+
+    this.db = db
+
+    this._open()
+
+    /** Drives being seeded at the moment */
+    this._drives = new Map()
+
+    goodbye(() => {
+      console.log('Gracefully closing Hyperswarm node')
+      this.swarm.destroy()
+    })
+  }
+
+  async _open () {
+    for await (const [url] of this.db.iterator()) {
+      this._seed(url)
+    }
+  }
+
+  /**
+   * @param {string} url
+   */
+  async _seed (url) {
+    if (this._drives.has(url)) return
+
+    const key = SlashURL.parse(url).key
+
+    const drive = new HyperDrive(this.corestore, key)
+    await drive.ready()
+
+    this._drives.set(url, drive)
+
+    // TODO: Only to connect to seeders, remove if/when we get rid of seeders topic
+    this.swarm.join(SEEDERS_TOPIC, { server: false, client: true })
```

Comment:
`await this.swarm.join(SEEDERS_TOPIC, { server: false, client: true }).flushed()` ?

or will it flush both on the next step?

Reply:
`flushed()` awaits nothing unless you are joining in server mode